src/java/org/apache/cassandra/index/sai/analyzer/LuceneAnalyzer.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * All changes to the original code are Copyright DataStax, Inc.
+ *
+ * Please see the included license file for details.
+ */
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.analyzer;
+
+import java.io.CharArrayReader;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.Map;
+
+import com.google.common.base.MoreObjects;
+
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.index.sai.disk.io.BytesRefUtil;
+import org.apache.cassandra.utils.Throwables;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.tokenattributes.TermToBytesRefAttribute;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.BytesRefBuilder;
+import org.apache.lucene.util.CharsRef;
+import org.apache.lucene.util.CharsRefBuilder;
+
+public class LuceneAnalyzer extends AbstractAnalyzer
+{
+    public static final String JSON_ANALYZER = "json_analyzer";
+    public static final String JSON_QUERY_ANALYZER = "json_query_analyzer";
+    private AbstractType<?> type;
+    private boolean hasNext = false;
+
+    private final Analyzer analyzer;
+    private TokenStream tokenStream;
+    private final CharsRefBuilder charsBuilder = new CharsRefBuilder();
+    private TermToBytesRefAttribute termAttr;
+    private final BytesRefBuilder bytesBuilder = new BytesRefBuilder();
+    private final Map<String, String> options;
+
+    public LuceneAnalyzer(AbstractType<?> type, Analyzer analyzer, Map<String, String> options) throws Exception
+    {
+        this.type = type;
+        this.analyzer = analyzer;
+        this.options = options;
+    }
+
+    @Override
+    public void close() throws IOException
+    {
+        if (analyzer != null) analyzer.close();
+    }
+
+    @Override
+    public boolean hasNext()
+    {
+        try
+        {
+            hasNext = tokenStream.incrementToken();
+
+            if (hasNext)
+            {
+                final BytesRef br = termAttr.getBytesRef();
+
+                // TODO: reuse next
```

Comment:
Not sure how it can be reused since we need to create a new `ByteBuffer` each time.

Reply:
Correct